test/python/test_tools_import_special_phrases.py

```diff
@@ -0,0 +1,346 @@
+"""
+    Tests for import special phrases methods
+    of the class SpecialPhrasesImporter.
+"""
+from nominatim.errors import UsageError
+from pathlib import Path
+import tempfile
+from shutil import copyfile
+import pytest
+from nominatim.tools.special_phrases import SpecialPhrasesImporter
+
+TEST_BASE_DIR = Path(__file__) / '..' / '..'
+
+def test_check_sanity_class(special_phrases_importer):
+    """
+        Check for _check_sanity() method.
+        If a wrong class or type is given, an UsageError should raise.
+        If a good class and type are given, nothing special happens.
+    """
+    with pytest.raises(UsageError) as wrong_class:
+        special_phrases_importer._check_sanity('en', '', 'type')
+    
+    with pytest.raises(UsageError) as wrong_type:
+        special_phrases_importer._check_sanity('en', 'class', '')
+
+    special_phrases_importer._check_sanity('en', 'class', 'type')
+
+    assert wrong_class and wrong_type
+
+def test_load_white_and_black_lists(special_phrases_importer):
+    """
+        Test that _load_white_and_black_lists() well return
+        black list and white list and that they are of dict type.
+    """
+    black_list, white_list = special_phrases_importer._load_white_and_black_lists()
+
+    assert isinstance(black_list, dict) and isinstance(white_list, dict)
+
+def test_convert_php_settings(special_phrases_importer):
+    """
+        Test that _convert_php_settings_if_needed() convert the given
+        php file to a json file.
+    """
+    php_file = (TEST_BASE_DIR / 'testfiles' / 'phrase_settings.php').resolve()
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_settings = (Path(temp_dir) / 'phrase_settings.php').resolve()
+        copyfile(php_file, temp_settings)
+        special_phrases_importer._convert_php_settings_if_needed(temp_settings)
+
+        assert (Path(temp_dir) / 'phrase_settings.json').is_file()
+
+def test_convert_settings_wrong_file(special_phrases_importer):
+    """
+        Test that _convert_php_settings_if_needed() raise an exception
+        if the given file is not a valid file.
+    """
+
+    with pytest.raises(UsageError) as exceptioninfos:
+        special_phrases_importer._convert_php_settings_if_needed('random_file')
+
+    assert str(exceptioninfos.value) == 'random_file is not a valid file.'
```

Comment:
Minor nitpicking on the use of `pytest.raises`: you don't need an explicit assert. Just the `with pytest.raises()` expression will do. It includes the assert. And if you want to check for a certain message in your exception, use the [match parameter](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions).